shared_model/backend/protobuf/queries/proto_query.hpp

```diff
@@ -81,6 +81,17 @@ namespace shared_model {
 
       interface::types::TimestampType createdTime() const override;
 
+    #ifdef EMSCRIPTEN
```

Comment:
I tried for days to solve this issue. 
My first principle during implementation was to completely separate emscripten library logic from the bindings, but I didn't find a solution for this.

In short, the problem is:
We have the methods return simple types like uint64_t. This types cannot be cast to js types automatically by Embind. So, we can't create some kind of js wrapper for such types before they will be converted on the C++ side to something Embind can consume.
Also, we can't simply create child class in C++ side and bind it instead of a binding class. Because our API design, in any case, will return an instance of a binding class. To create an appropriate descendant we at least have to overload all the methods.

So this is the less painful solution then others I found. Adding this conditional inclusion was briefly discussed with @muratovv .

Also, I'm not a good C++ specialist, so if you can suggest some tricky trick to solve this it will be awesome.